mdoc/src/main/scala/mdoc/internal/markdown/Instrumenter.scala

```diff
@@ -122,6 +123,7 @@ object Instrumenter {
     def binders(pat: Pat): List[Name] =
       pat.collect { case m: Member => m.name }
     def unapply(tree: Tree): Option[List[Name]] = tree match {
+      case Defn.Val(mods, _, _, _) if mods.exists(_.isInstanceOf[Lazy]) => Some(Nil)
```

Comment:
While this fix avoids the eager evaluation, it would be nice to print out the type of the variable like in the Scala REPL
```scala
scala> lazy val x = 42
x: Int = <lazy>
```

I think it might be possible to achieve this by passing in a custom `_root_.sourcecode.Text(binder, "<lazy>")` in the `printBinders` method only when the definition has `Mod.Lazy`.

Reply:
Sure ! I can give it a go 👍

Reply:
I haven't managed to implement the improvement unfortunately, it not as simple as I thought it would be / I need to spend more time on it. So I'll suggest to merge this PR, if raise an another issue for the improvement.

Reply:
Sounds good!